WalletWasabi/WabiSabi/Backend/Banning/CoinVerifier.cs

```diff
@@ -43,15 +44,15 @@ internal CoinVerifier(CoinJoinIdStore coinJoinIdStore, CoinVerifierApiClient api
 	public CoinVerifierLogger VerifierAuditArchiver { get; }
 
 	private CoinVerifierApiClient CoinVerifierApiClient { get; }
-	private ConcurrentDictionary<Coin, CoinVerifyItem> CoinVerifyItems { get; } = new();
+	private ConcurrentDictionary<Coin, CoinVerifyItem> CoinVerifyItems { get; } = new(CoinEqualityComparer.Default);
 
 	public async Task<IEnumerable<CoinVerifyResult>> VerifyCoinsAsync(IEnumerable<Coin> coinsToCheck, CancellationToken cancellationToken)
 	{
 		using CancellationTokenSource timeoutCancellationTokenSource = new(TimeSpan.FromSeconds(30));
 		using CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(timeoutCancellationTokenSource.Token, cancellationToken);
 
 		// Booting up the results with the default value - ban: no, remove: yes.
-		Dictionary<Coin, CoinVerifyResult> coinVerifyItems = coinsToCheck.ToDictionary(coin => coin, coin => new CoinVerifyResult(coin, ShouldBan: false, ShouldRemove: true));
+		Dictionary<OutPoint, CoinVerifyResult> coinVerifyItems = coinsToCheck.ToDictionary(coin => coin.Outpoint, coin => new CoinVerifyResult(coin, ShouldBan: false, ShouldRemove: true));
```

Comment:
We could pass the new `CoinEqualityComparer.Default` here, too.

```suggestion
		Dictionary<Coin, CoinVerifyResult> coinVerifyItems = coinsToCheck.ToDictionary(coin => coin, coin => new CoinVerifyResult(coin, ShouldBan: false, ShouldRemove: true), CoinEqualityComparer.Default);
```

But either way, it looks good.

Reply:
@adamPetho ok, i will do that just to respect the original design. However, take into account the this CoinEqualityComparer is an ugliness to solve a potential problem without touching too much code.

@kiminuo yes, you're right. It can be done in a separate PR, i don't want to touch more. It is more of less safe to assume the coins will be unique.

Reply:
done

Reply:
It wasn't mandatory, only a suggestion.